tests/bin/test.sh

```diff
@@ -5,25 +5,22 @@
 set -eo pipefail
 
 function test_missing_properties() {
-	./node_modules/node-jq/bin/jq -e '(.required // []) as $req | (.properties // []) as $props | ($req - ($req | map(select(. as $r | $props | has($r)))) | length) == 0' "$1" > /dev/null \
-		|| ( echo "Properties in the 'required' element of $1 are missing from the 'properties' element" && exit 1 )
+	local file="$1"
+	if [ "[]" != "$(./node_modules/node-jq/bin/jq -r '(.required // []) as $req | (.properties // []) as $props | $req - ($req | map(select(. as $r | $props | has($r))))' "$file")" ]
+	then
+		echo "Properties in the 'required' element of ${file} are missing from the 'properties' element" 1>&2
+		exit 1
+	fi
 }
 
 function validate_schema() {
 	local file="$1"
-	local base=${file//schemas\//}
-	local base=${base/.json/}
-	./node_modules/.bin/ajv validate -m tests/hyper-schema/hyper-schema.json -r "schemas/**/*.json" -s "$file" -d "tests/data/$base/*.json"
+	local base="$(basename "$file" .json)"
+	./node_modules/.bin/ajv validate -m tests/hyper-schema/hyper-schema.json -r "schemas/**/*.json" -s "$file" -d "tests/data/${base}/*.json"
```

Comment:
Pretty sure that using `basename` here means the path provided to the `-d` flag is incorrect for the REST API tests

Reply:
I see! That "base" is not the basename.
```bash
$ basename schemas/rest-api/post.json .json
post
```

Reply:
It should be "rest-api/post"

Reply:
Why have ajv validate not failed??

Reply:
Only `tests/data/rest-api/routes/*.json` files exist, and `schemas/rest-api/routes.json` does **not** exist.
What do you think?

Reply:
I'm working on the exception as we speak :) You might want to hold off making any further changes to this file while I work on that.